src/components/App.jsx

```diff
@@ -1,32 +1,44 @@
 import React, { Component } from 'react';
-import { ChangePageWrapper } from './App.Styles';
+import { Provider } from 'react-redux';
+import store from '../redux/store';
 import GlobalStyle from './GlobalStyle';
 import ErrorBoundary from './ErrorBoundary';
 import ChangePageButton from './Helper/ChangePageButton/ChangePageButton';
 import SearchPage from './Pages/SearchPage/SearchPage';
 import DetailPage from './Pages/DetailPage/DetailPage';
 
+import { ChangePageWrapper } from './App.Styles';
+
 class App extends Component {
   state = {
-    pageType: 'search'
+    pageType: 'search',
```

Comment:
This variable is redundant. You can always calculate it on the fly using selectedMovie (if selectedMovie is presented - show detail, otherwise show search).